make.js

```diff
@@ -577,3 +578,112 @@ target.bump = function() {
         }
     });
 }
+
+// Generate spritly zip
+// This methods generate a zip file that contains the tip of all task major versions for the last sprint
+// Use:
+//   node make.js gensprintlyzip --sprint=m153 --outputdir=E:\testing\ --depxmlpath=C:\Users\stfrance\Desktop\tempdeps.xml
+// 
+// Result:
+//   A zip file named a.b.x.zip
+// 
+// The generated zip can be uploaded to an account using tfx cli and it will install all of the tasks contained in the zip.
+// The zip should be uploaded to the azure-pipelines-tasks repository
+// 
+// Process:
+// 
+//  We create a workspace folder to do all of our work in. This is created in the output directory. output-dir/workspace-GUID
+//  Inside here, we first create a package file based on the packages we want to download.
+//  Then nuget restore, then get zips, then create zip.
+target.gensprintlyzip = function() {
+    var sprint = options.sprint;
+    var outputDirectory = options.outputdir;
+    var dependenciesXmlFilePath = options.depxmlpath;
+    var taskFeedUrl = 'https://mseng.pkgs.visualstudio.com/_packaging/Codex-Deps/nuget/v3/index.json';
+
+    console.log('# Creating sprintly zip.');
+
+    console.log('\n# Loading tasks from dependencies file.');
+    var dependencies = fs.readFileSync(dependenciesXmlFilePath, 'utf8');
+
+    var dependenciesArr = dependencies.split('\n');
+    console.log(`Found ${dependenciesArr.length} dependencies.`);
+
+    var taskDependencies = [];
+    var taskStringArr = [];
+
+    dependenciesArr.forEach(function (currentDep) {
+        if (currentDep.indexOf('Mseng.MS.TF.DistributedTask.Tasks.') === -1) {
+            return;
+        }
+
+        taskStringArr.push(currentDep);
+
+        var depDetails = currentDep.split("\"");
+        var name = depDetails[1];
+        var version = depDetails[3];
+
+        taskDependencies.push({ 'name': name, 'version': version });
+    });
+
+    console.log(`Found ${taskDependencies.length} task dependencies.`);
+
+    console.log('\n# Downloading task nuget packages.');
+
+    var tempWorkspaceDirectory = `${outputDirectory}\\workspace-${Math.floor(Math.random() * 1000000000)}`;
+    console.log(`Creating temporary workspace directory ${tempWorkspaceDirectory}`);
+
+    fs.mkdirSync(tempWorkspaceDirectory);
+
+    console.log('Writing packages.config file');
+
+    var packagesConfigPath = `${tempWorkspaceDirectory}\\packages.config`;
+    var packagesConfigContent = '<?xml version="1.0" encoding="utf-8"?>\n';
+    packagesConfigContent += '<packages>\n';
+
+    taskStringArr.forEach(function (taskString) {
+        packagesConfigContent += taskString;
```

Comment:
Should these be separated by `\n`?

Reply:
The line already contains newline character.

Reply:
I'm a little confused by this. Isn't the basic flow (with reduced logic):
```
var dependenciesArr = dependencies.split('\n');
var taskStringArr = [];
dependenciesArr.forEach(function (currentDep) {
    taskStringArr.push(currentDep);
}
taskStringArr.forEach(function (taskString) {
    packagesConfigContent += taskString;
}
```

Wouldn't the `\n` characters get removed when we split on it?

Reply:
Oh yeah it's possible, the file is intermittent (I use it to restore packages then delete it) so we should be OK. I will take a look.